File: backend-service/src/main/java/az/schedule/backendservice/service/AnalyticsService.java
package az.schedule.backendservice.service;

import az.schedule.backendservice.dto.response.analytics.AIInsights;
import az.schedule.backendservice.dto.response.analytics.TaskAnalyticsResponse;

public interface AnalyticsService {
    /**
     * Get task analytics without AI insights (fast)
     * @param accountId User account ID
     * @return Analytics data with stats, charts, heatmap (AI insights will be null)
     */
    TaskAnalyticsResponse getTaskAnalytics(Long accountId);
    
    /**
     * Get AI-powered insights separately (slower)
     * @param accountId User account ID
     * @return AI-generated insights and suggestions
     */
    AIInsights getAIInsights(Long accountId);
}

File: FE-AzSchedule/src/services/analyticsService.js
import axiosInstance from '../lib/axios';

/**
 * Get task analytics (fast - stats, charts, heatmap)
 * @returns {Promise} Analytics data without AI insights
 */
export const getTaskAnalytics = async () => {
  const response = await axiosInstance.get('/analytics');
  return response.data;
};

/**
 * Get AI-powered insights (slower - loaded separately)
 * @returns {Promise} AI insights and suggestions
 */
export const getAIInsights = async () => {
  const response = await axiosInstance.get('/analytics/ai-insights');
  return response.data;
};
